### ISSUE TITLE
aiplatform.ExperimentRun.list fails if user manually deletes a TensorBoard

### ISSUE THREAD

**Issue by jasonbrancazio** [googleapis/python-aiplatform#2299]
This occurred in v 1.26.0:

If a user uses aiplatform.init() with an experiment name to create an experiment, and then aiplatform.start_run to start an experiment_run, then deletes the TensorBoard using the UI, aiplatform.ExperimentRun.list will report this error:

_InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.NOT_FOUND
	details = "The Tensorboard does not exist."

**Comment by zbl94** [comment]
Hi jasonbrancazio, 

Noticed the issue was created 2 weeks ago. Do you want to still delete the TensorBoard with the clarification in https://github.com/googleapis/python-aiplatform/issues/2298? If so,  would you mind share the sample code and full stack trace?

**Comment by jasonbrancazio** [comment]
I'm less likely to delete the TensorBoard, but given that it is possible, and leads to a bug, here are the steps to reproduce and the full stacktrace.

```
from google.cloud import aiplatform
aiplatform.init(project='my-project',location='us-central1',experiment='test-experiment')
aiplatform.start_run(run='test-experiment-run')
# this next line with variables filled in is output by aiplatform.init
tb = aiplatform.Tensorboard('projects/[MY_PROJECT_ID]/locations/us-central1/tensorboards/[TENSORBOARD_ID]')
tb.delete()
aiplatform.ExperimentRun.list(experiment='test-experiment')
```

```
---------------------------------------------------------------------------
_InactiveRpcError                         Traceback (most recent call last)
File .venv/lib/python3.10/site-packages/google/api_core/grpc_helpers.py:72, in _wrap_unary_errors.<locals>.error_remapped_callable(*args, **kwargs)
     71 try:
---> 72     return callable_(*args, **kwargs)
     73 except grpc.RpcError as exc:

File .venv/lib/python3.10/site-packages/grpc/_channel.py:1030, in _UnaryUnaryMultiCallable.__call__(self, request, timeout, metadata, credentials, wait_for_ready, compression)
   1028 state, call, = self._blocking(request, timeout, metadata, credentials,
   1029                               wait_for_ready, compression)
-> 1030 return _end_unary_response_blocking(state, call, False, None)

File .venv/lib/python3.10/site-packages/grpc/_channel.py:910, in _end_unary_response_blocking(state, call, with_call, deadline)
    909 else:
--> 910     raise _InactiveRpcError(state)

_InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.NOT_FOUND
        details = "The Tensorboard does not exist."
        debug_error_string = "UNKNOWN:Error received from peer ipv6:%5B2607:f8b0:4007:811::200a%5D:443 {grpc_message:"The Tensorboard does not exist.", grpc_status:5, created_time:"2023-06-27T11:17:08.411511-07:00"}"
>

The above exception was the direct cause of the following exception:

NotFound                                  Traceback (most recent call last)
Cell In[85], line 1
----> 1 aiplatform.ExperimentRun.list(experiment='test-experiment')

File .venv/lib/python3.10/site-packages/google/cloud/aiplatform/metadata/experiment_run_resource.py:536, in ExperimentRun.list(cls, experiment, project, location, credentials)
    529 with concurrent.futures.ThreadPoolExecutor(
    530     max_workers=max([len(run_contexts), len(run_executions)])
    531 ) as executor:
    532     submissions = [
    533         executor.submit(_initialize_experiment_run, context)
    534         for context in run_contexts
    535     ]
--> 536     experiment_runs = [submission.result() for submission in submissions]
    538     submissions = [
    539         executor.submit(_initialize_v1_experiment_run, execution)
    540         for execution in run_executions
    541     ]
    543     for submission in submissions:

File .venv/lib/python3.10/site-packages/google/cloud/aiplatform/metadata/experiment_run_resource.py:536, in <listcomp>(.0)
    529 with concurrent.futures.ThreadPoolExecutor(
    530     max_workers=max([len(run_contexts), len(run_executions)])
    531 ) as executor:
    532     submissions = [
    533         executor.submit(_initialize_experiment_run, context)
    534         for context in run_contexts
    535     ]
--> 536     experiment_runs = [submission.result() for submission in submissions]
    538     submissions = [
    539         executor.submit(_initialize_v1_experiment_run, execution)
    540         for execution in run_executions
    541     ]
    543     for submission in submissions:

File ~/.pyenv/versions/3.10.9/lib/python3.10/concurrent/futures/_base.py:458, in Future.result(self, timeout)
    456     raise CancelledError()
    457 elif self._state == FINISHED:
--> 458     return self.__get_result()
    459 else:
    460     raise TimeoutError()

File ~/.pyenv/versions/3.10.9/lib/python3.10/concurrent/futures/_base.py:403, in Future.__get_result(self)
    401 if self._exception:
    402     try:
--> 403         raise self._exception
    404     finally:
    405         # Break a reference cycle with the exception in self._exception
    406         self = None

File ~/.pyenv/versions/3.10.9/lib/python3.10/concurrent/futures/thread.py:58, in _WorkItem.run(self)
     55     return
     57 try:
---> 58     result = self.fn(*self.args, **self.kwargs)
     59 except BaseException as exc:
     60     self.future.set_exception(exc)

File .venv/lib/python3.10/site-packages/google/cloud/aiplatform/metadata/experiment_run_resource.py:505, in ExperimentRun.list.<locals>._initialize_experiment_run(context)
    502 this_experiment_run._metadata_node = context
    504 with experiment_resources._SetLoggerLevel(resource):
--> 505     tb_run = this_experiment_run._lookup_tensorboard_run_artifact()
    506 if tb_run:
    507     this_experiment_run._backing_tensorboard_run = tb_run

File .venv/lib/python3.10/site-packages/google/cloud/aiplatform/metadata/experiment_run_resource.py:382, in ExperimentRun._lookup_tensorboard_run_artifact(self)
    376         tensorboard_run_artifact = None
    378 if tensorboard_run_artifact and self._is_backing_tensorboard_run_artifact(
    379     tensorboard_run_artifact
    380 ):
    381     return experiment_resources._VertexResourceWithMetadata(
--> 382         resource=tensorboard_resource.TensorboardRun(
    383             tensorboard_run_artifact.metadata[
    384                 constants.GCP_ARTIFACT_RESOURCE_NAME_KEY
    385             ]
    386         ),
    387         metadata=tensorboard_run_artifact,
    388     )

File .venv/lib/python3.10/site-packages/google/cloud/aiplatform/tensorboard/tensorboard_resource.py:640, in TensorboardRun.__init__(self, tensorboard_run_name, tensorboard_id, tensorboard_experiment_id, project, location, credentials)
    630     raise ValueError(
    631         "Both tensorboard_id and tensorboard_experiment_id must be provided or neither should be provided."
    632     )
    634 super().__init__(
    635     project=project,
    636     location=location,
    637     credentials=credentials,
    638     resource_name=tensorboard_run_name,
    639 )
--> 640 self._gca_resource = self._get_gca_resource(
    641     resource_name=tensorboard_run_name,
    642     parent_resource_name_fields={
    643         Tensorboard._resource_noun: tensorboard_id,
    644         TensorboardExperiment._resource_noun: tensorboard_experiment_id,
    645     }
    646     if tensorboard_id
    647     else tensorboard_id,
    648 )
    650 self._time_series_display_name_to_id_mapping = (
    651     self._get_time_series_display_name_to_id_mapping()
    652 )

File .venv/lib/python3.10/site-packages/google/cloud/aiplatform/base.py:643, in VertexAiResourceNoun._get_gca_resource(self, resource_name, parent_resource_name_fields)
    623 """Returns GAPIC service representation of client class resource.
    624 
    625 Args:
   (...)
    630         Should not include project and location.
    631 """
    632 resource_name = utils.full_resource_name(
    633     resource_name=resource_name,
    634     resource_noun=self._resource_noun,
   (...)
    640     resource_id_validator=self._resource_id_validator,
    641 )
--> 643 return getattr(self.api_client, self._getter_method)(
    644     name=resource_name, retry=_DEFAULT_RETRY
    645 )

File .venv/lib/python3.10/site-packages/google/cloud/aiplatform_v1/services/tensorboard_service/client.py:2277, in TensorboardServiceClient.get_tensorboard_run(self, request, name, retry, timeout, metadata)
   2272 metadata = tuple(metadata) + (
   2273     gapic_v1.routing_header.to_grpc_metadata((("name", request.name),)),
   2274 )
   2276 # Send the request.
-> 2277 response = rpc(
   2278     request,
   2279     retry=retry,
   2280     timeout=timeout,
   2281     metadata=metadata,
   2282 )
   2284 # Done; return the response.
   2285 return response

File .venv/lib/python3.10/site-packages/google/api_core/gapic_v1/method.py:113, in _GapicCallable.__call__(self, timeout, retry, *args, **kwargs)
    110     metadata.extend(self._metadata)
    111     kwargs["metadata"] = metadata
--> 113 return wrapped_func(*args, **kwargs)

File .venv/lib/python3.10/site-packages/google/api_core/retry.py:349, in Retry.__call__.<locals>.retry_wrapped_func(*args, **kwargs)
    345 target = functools.partial(func, *args, **kwargs)
    346 sleep_generator = exponential_sleep_generator(
    347     self._initial, self._maximum, multiplier=self._multiplier
    348 )
--> 349 return retry_target(
    350     target,
    351     self._predicate,
    352     sleep_generator,
    353     self._timeout,
    354     on_error=on_error,
    355 )

File .venv/lib/python3.10/site-packages/google/api_core/retry.py:191, in retry_target(target, predicate, sleep_generator, timeout, on_error, **kwargs)
    189 for sleep in sleep_generator:
    190     try:
--> 191         return target()
    193     # pylint: disable=broad-except
    194     # This function explicitly must deal with broad exceptions.
    195     except Exception as exc:

File .venv/lib/python3.10/site-packages/google/api_core/grpc_helpers.py:74, in _wrap_unary_errors.<locals>.error_remapped_callable(*args, **kwargs)
     72     return callable_(*args, **kwargs)
     73 except grpc.RpcError as exc:
---> 74     raise exceptions.from_grpc_error(exc) from exc

NotFound: 404 The Tensorboard does not exist.
```

**Comment by serop-ba** [comment]
any update on this ? I don't want to create the costly Tensorboard by default

**Comment by lakeyk** [comment]
The issue of running an Experiment with a deleted Tensorboard was fixed with a recent update and should be reflected in the next SDK release. 

The option to not create and use a default Tensorboard has been available since [version 1.36.3](https://github.com/googleapis/python-aiplatform/releases/tag/v1.36.3).
This can be done by setting `experiment_tensorboard=False` in the `aiplatform.init` call
